ros2/launch.sh.tpl

```diff
@@ -2,6 +2,13 @@
 
 set -o errexit -o nounset -o pipefail
 
+if [ -z "${ROS_HOME:-}" ] && [ -z "${ROS_LOG_DIR:-}" ]; then
```

Comment:
For relation(s) between ROS_HOME and ROS_LOG_DIR folks should read ROS docs, not docs/impl of this repo. Here we should tend to be as general as possible and avoid introducing hidden functionality. Therefore, I'd ask again for separate setup of those two env vars like I explained earlier.

Next to this, this piece of code is only relevant for tests. Thus, this should look like (for better explicitness IMO):
```sh
if [[ ! -z "${BAZEL_TEST}" ]]; then
  bazel_test_output_dir="${TEST_UNDECLARED_OUTPUTS_DIR:-${TEST_TMPDIR}}"
  if [[ -z "${ROS_HOME:-}" ]]; then
    export ROS_HOME="${bazel_test_output_dir}" 
  fi
  if [[ -z "${ROS_LOG_DIR:-}" ]]; then
    export ROS_LOG_DIR="${bazel_test_output_dir}" 
  fi
fi
```